microservices/services/contest-service/pom.xml

```diff
@@ -105,6 +105,9 @@
                                 | MembershipRequest | xcolab_PointType
                                 | xcolab_ProposalReference | xcolab_ProposalRatingValue
                                 | xcolab_ProposalRatingType | xcolab_PointsDistributionConfiguration
+                                | comment_Comment | comment_Thread
+                                | activities_ActivityEntry | xcolab_ActivitySubscription
+                                | Users_Groups
```

Comment:
These tables are owned by different microservices - we should not be accessing them (let alone delete from them) from the contest microservice. I assume you did this to allow a transaction across all these tables, which is a nice goal but the separation of responsibilities is more important.

* `activities_ActivityEntry` and `xcolab_ActivitySubscription` should be deleted in the activities-service. Deleting them *before* deleting the contest/proposal/comment they reference does no harm, so they should be deleted first.
* `comment_Comment` and `comment_Thread` should be deleted in one transaction in the comment-service. As there are no other references to them, deleting them *after* the contest referencing them is deleted should do no harm.
* `Users_Groups` should be deleted in the members-service. As nothing references them, having groups stick around after the proposal referencing them should do no harm, so this method can be called *after* the proposal(s) is/are deleted.

For groups and comments, it would probably be nice to use the same idea you used for proposals: deleting orphaned threads/groups all at once (the first time we call this method, it will delete all the previously orphaned entries, which is a nice plus).

Reply:
Done. I've moved the deletions to their respective services. It is not as simple as just doing it before and after, because we need intermediate results to know what to delete. The contest-service now calls the three other services to delete the linked data.

I'm against reusing approach with deleting orphaned entries for two reasons:
* It's not as trivial to say whether a group or a comment is actually orphaned. I don't know all the possible links there.
* The `deleteContest` method should not have too many unexpected side effects. While it may sound nice now that we "clean up the database" as a side effect, it is not a good coding standard and may lead to issues down the road. Imagine someone introduces a new place to have discussions. Since we don't check for that relation in the code here, all the threads/comments there would be deleted if someone executes a delete mass action.